src/containers/BooksForm.js

```diff
@@ -0,0 +1,112 @@
+/* eslint-disable no-underscore-dangle */
+/* eslint-disable jsx-a11y/label-has-associated-control */
+import React, { useState } from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import { createBook } from '../actions';
+
+const categories = [
+  'Action',
+  'Biography',
+  'History',
+  'Horror',
+  'Kids',
+  'Learning',
+  'Sci-Fi',
+];
+
+const mapStateToProps = ({ books }) => ({ books });
+const mapDispatchToProps = dispatch => ({
+  createBook: book => dispatch(createBook(book)),
+});
+
+const BooksForm = ({ createBook }) => {
+  const [errors, setErrors] = useState({});
+  const [book, setBook] = useState({
+    id: Math.floor(Math.random() * 1000),
+    title: '',
+    category: '',
+  });
+
+  const handleChange = ({ target }) => {
+    setBook({
+      ...book,
+      [target.name]: target.value,
+    });
+  };
+
+  const formIsValid = () => {
+    const _errors = {};
+
+    if (!book.title) _errors.title = 'Title is required';
+    if (!book.category) _errors.category = 'Category is required';
+
+    setErrors(_errors);
+    return Object.keys(_errors).length === 0;
+  };
+
+  const handleSubmit = event => {
+    event.preventDefault();
+    if (!formIsValid()) return;
+
+    createBook({ id: book.id, title: book.title, category: book.category });
+    book.title = '';
+    book.category = '';
```

Comment:
- [ ] States cannot be manipulated directly. You will need to use `setBook` here.